pkg/generators/generator.go

```diff
@@ -37,12 +37,11 @@ type TokenIndex uint64
 
 type DistributionFunc func() TokenIndex
 
-type GeneratorInterface interface {
+type Interface interface {
```

Comment:
Newbie question: verbosity with Structs/Interfaces should happen only when you have more than one of these inside a package?

IMHO keep the GeneratorInterface is way better for readability.

Reply:
There is need for that, as package name is `generators` and usage in any other package would look like this:

```go
import (
   "generators"
)

func a(gen generators.Interface) {}
```

As you can see, problem with readability, is solved with a package name